src/axom/inlet/tests/inlet_object.cpp

```diff
@@ -939,6 +939,69 @@ TYPED_TEST(inlet_object, struct_arrays_as_std_vector)
   EXPECT_EQ(foos, expected_foos);
 }
 
+TYPED_TEST(inlet_object, default_scalar_marked_false)
+{
+  std::string testString = " ";
+  DataStore ds;
+  Inlet inlet = createBasicInlet<TypeParam>(&ds, testString);
+
+  auto& scalar = inlet.addInt("foo").defaultValue(2);
```

Comment:
It might also be good to test how this new feature interoperates with `.required()`